Mathlib/Analysis/Convex/Schur.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2025 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+import Mathlib.Order.Monotone.Monovary
+
+/-!
+# Schur's inequality
+
+This file proves a number of forms of Schur's inequality.
+The proofs are adapted from https://www.cip.ifi.lmu.de/~grinberg/VornicuS.pdf
+-/
+
+lemma vornicu_schur_mildorf {α : Type*} [PartialOrder α] [CommRing α] [IsOrderedRing α]
+    {a b c x y z : α}
+    (hx : 0 ≤ x) (hz : 0 ≤ z)
+    (hcb : c ≤ b) (hba : b ≤ a)
+    (hyxz : y ≤ x + z) :
+    0 ≤ x * (a - b) * (a - c) + y * (b - a) * (b - c) + z * (c - a) * (c - b) := by
+  have h₁ : 0 ≤ a - b := by linear_combination hba
+  have h₂ : 0 ≤ b - c := by linear_combination hcb
+  have h₃ : 0 ≤ x + z - y := by linear_combination hyxz
+  have h₄ : 0 ≤ x * (a - b) := mul_nonneg hx h₁
+  calc
+    _ ≤ (x + z - y) * (b - c) * (a - b) := mul_nonneg (mul_nonneg h₃ h₂) h₁
+    _ = x * (a - b) * (b - c) + y * (b - a) * (b - c) + z * (a - b) * (b - c) := by ring
+    _ ≤ x * (a - b) * (a - c) + y * (b - a) * (b - c) + z * (a - c) * (b - c) := by gcongr
+    _ = x * (a - b) * (a - c) + y * (b - a) * (b - c) + z * (c - a) * (c - b) := by ring
+
+lemma generalized_schur {α : Type*} [LinearOrder α] [CommRing α] [IsOrderedRing α]
+    {a b c x y z : α}
+    (hx : 0 ≤ x) (hy : 0 ≤ y) (hz : 0 ≤ z)
+    (h : Monovary ![a, b, c] ![x, y, z]) :
```

Comment:
For this to be reasonable for the caller I think we need a `monovary_vecCons_iff` simp lemma.

Reply:
In all honesty I think this one isn't that reasonable, most of the time in practice you'd have the variables already ordered. I'll add a comment to that effect, and try to make the lemma you describe nonetheless.